src/kubernetes.cc

```diff
@@ -199,13 +209,18 @@ json::value KubernetesReader::ComputePodAssociations(const json::Object* pod)
     node_name = json::string(spec->Get<json::String>("nodeName"));
   }
 
-  // If controllers or node_name are not populated, they will be discarded.
+  // If any of the field values are not populated, they will be discarded.
```

Comment:
If the instance was not found, it makes sense to not have the entry for "infrastructureResource"

Reply:
But that's exactly what's going to happen. If the instance was not found, `instance_resource` will be `nullptr`, which will cause the "infrastructureResource" field to be discarded. I can see how the new comment can be confusing, so I've changed it to be more precise. Is this better?

Reply:
Thanks - this looks better.